src/com/facebook/buck/features/go/GoCompile.java

```diff
@@ -278,7 +279,7 @@ public GoCompile(
     List<Path> srcFiles = new ArrayList<>();
     for (SourcePath path : srcPaths) {
       Path srcPath = context.getSourcePathResolver().getAbsolutePath(path);
-      if (Files.isDirectory(srcPath)) {
+      if (path instanceof BuildTargetSourcePath && Files.isDirectory(srcPath)) {
```

Comment:
@linzhp why is this needed? after I removed the instanceof the tests are still passing, if this is really required can you update the test to reflect the case?:
`path instanceof BuildTargetSourcePath`

Reply:
thanks for explanation. @styurin can you check that line? If that is the case likely we might need to update other places as well